generateFiles/end_to_end_api_test.go

```diff
@@ -0,0 +1,569 @@
+package generateFiles
+
+import (
+	"encoding/csv"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	mgo "gopkg.in/mgo.v2"
+
+	"github.com/ONSdigital/dp-api-tests/testDataSetup/mongo"
+	"github.com/ONSdigital/dp-api-tests/testDataSetup/neo4j"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/gavv/httpexpect"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestSuccessfulEndToEndProcess(t *testing.T) {
+
+	importAPI := httpexpect.New(t, cfg.ImportAPIURL)
+	recipeAPI := httpexpect.New(t, cfg.RecipeAPIURL)
+	datasetAPI := httpexpect.New(t, cfg.DatasetAPIURL)
+	filterAPI := httpexpect.New(t, cfg.FilterAPIURL)
+
+	hasRemovedAllResources := true
+	filename := "v4TestFile.csv"
+	recipe := "2943f3c5-c3f1-4a9a-aa6e-14d21c33524c"
+
+	// Get dataset ID from recipe API
+	recipeResponse := recipeAPI.GET("/recipes/{recipe}", recipe).
+		Expect().Status(http.StatusOK).JSON().Object()
+
+	recipeResponse.Value("id").NotNull()
+
+	Convey("Given a v4 file exists in aws", t, func() {
+		// Send v4 file to aws
+		_, err := sendV4FileToAWS(region, bucketName, filename)
+		if err != nil {
+			os.Exit(1)
+		}
+
+		// import API expects a s3 url as the location of the file
+		location := "s3://" + bucketName + "/" + filename
+
+		Convey("When a job is imported and the version of the dataset is published", func() {
+
+			// STEP 1 - Create job with state created
+			postJobResponse := importAPI.POST("/jobs").WithBytes([]byte(createValidJobJSON(recipe, location))).
+				Expect().Status(http.StatusCreated).JSON().Object()
+
+			postJobResponse.Value("id").NotNull()
+			jobID := postJobResponse.Value("id").String().Raw()
```

Comment:
Why is `.Raw()` used here. Do you just need to call `postJobResponse.Value("id").String()`

Reply:
No because jobID should be a string and `postJobResponse.Value("id").String()` is a `*httpexpect.String` type. By using raw function changes the type to string.